backend/api/rooms/[roomId]/connectUser.js

```diff
@@ -0,0 +1,37 @@
+const handler = require('../../../handler/handler');
+const Host = require('../../../handler/Classes/Host');
+const Room = require('../../../handler/Classes/Room');
+
+const connectUser = async (req, res) => {
+
+    let data = handler.getData();
+
+    let response;
+
+    let room = Room.getRoomById(req.query.roomId, data.rooms);
+    if (room) {
+        if (req.body.myToken === room.host.myToken) {
+            response = {error: false};
+            res.status(200);
+        } else {
+            if (req.body.myToken) {
+                room.addUser(req.body.clientName, req.body.myToken);
+                response = {error: false};
+            } else {
+                let newUser = room.addUser(req.body.clientName, Host.createToken(20));
+                response = {error: false, myToken: newUser.myToken};
+            }
+
+            res.status(200);
+        }
+    } else {
+        response = {error: true, message: 'Room not found'};
+        res.status(400);
+    }
+
+    handler.setData(data);
```

Comment:
data isn't changed here either, check in other files too. this reduces the time for the request answer

Reply:
but it is

Reply:
`room.addUser(req.body.clientName, req.body.myToken);`

This line addes the user that just connected to the room.connectedUser array, this has to be stored else it will not display the connected users on the host sidebar.